### PR TITLE
Add message feeds for synchronization and framing

### PR DESCRIPTION
**Pull request by mhidalgo-bdai**
Pending on #97, #98, and #99.

### CODE REVIEW

File: bdai_ros2_wrappers/bdai_ros2_wrappers/filters.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def __init__(
         """
         super().__init__()
         self._logger = logger
-        self._lock = threading.Lock()
+        self._lock = threading.RLock()
```
Comment (mhidalgo-bdai) [inline bdai_ros2_wrappers/bdai_ros2_wrappers/filters.py:46]: A little bug that didn't pop up in #98

File: bdai_ros2_wrappers/bdai_ros2_wrappers/utilities.py [hunk 2]
```diff
@@ -450,3 +460,20 @@ def take_kwargs(func: Callable, kwargs: Mapping) -> Tuple[Mapping, Mapping]:
...
+    """Ensures `value` is not None or fails trying."""
+    if value is None:
+        frame = inspect.currentframe()
+        assert frame is not None
+        frame = frame.f_back
+        assert frame is not None
+        traceback = inspect.getframeinfo(frame)
+        message = f"{traceback.filename}:{traceback.lineno}: "
+        if traceback.code_context is not None:
+            message += "".join(traceback.code_context).strip("\n ") + " failed"
```
Comment (amessing-bdai) [inline bdai_ros2_wrappers/bdai_ros2_wrappers/utilities.py:475]: Should this traceback block of code be its own function? Seems generic enough.

File: bdai_ros2_wrappers/bdai_ros2_wrappers/feeds.py [hunk 3]
```diff
@@ -99,3 +101,105 @@ def stream(
...
+        *,
+        tolerance_sec: float = 1.0,
+        tf_buffer: Optional[tf2_ros.Buffer] = None,
+        history_length: Optional[int] = None,
+        node: Optional[Node] = None,
+    ) -> None:
+        """Initializes the message feed.
+
+        Args:
+            feed: the upstream message feed.
```
Comment (baxelrod-bdai) [inline bdai_ros2_wrappers/bdai_ros2_wrappers/feeds.py:122]: should we use `@param` to conform with Doxygen formatting?

File: bdai_ros2_wrappers/bdai_ros2_wrappers/utilities.py (hunk 2 above)
Comment (mhidalgo-bdai) [reply bdai_ros2_wrappers/bdai_ros2_wrappers/utilities.py]: Sure thing. Done in 8ab505fd9114ccda1942d56b2722983f993a5423.

File: bdai_ros2_wrappers/bdai_ros2_wrappers/feeds.py (hunk 3 above)
Comment (mhidalgo-bdai) [reply bdai_ros2_wrappers/bdai_ros2_wrappers/feeds.py:122]: As far as I've seen, we've been using [Google's styleguide](https://google.github.io/styleguide/pyguide.html) for Python docstrings everywhere. We could use Doxygen's, or Numpy's, or Sphinx's instead. I would do that in follow-up PRs though.

**Review by amessing-bdai — APPROVED** [review]

**Review by khughes-bdai — APPROVED** [review]
LGTM

**Pull request merged**